### PR TITLE
Handle unrouteable Table API requests.

### PR DESCRIPTION
**Pull request by mjkillough**
This PR does two things:
 - Handles unrouteable Table API requests.
 - Adds an API for getting distance from Table API requests.

It introduces an `Annotation` bitfield, that can be used like
`Annotations::All` / `Annotations::Duration | Annotations::Distance` /
`Annotations::Duration`, to specify what should be returned from OSRM.

### CODE REVIEW

File: src/table.rs [hunk 1]
```diff
@@ -8,8 +28,17 @@ pub struct Parameters {
...
 
 impl Parameters {
-    pub fn new() -> Result<Parameters> {
+    pub fn new(annotations: Annotations) -> Result<Parameters> {
         let handle = call_with_error!(osrmc_table_params_construct())?;
+
+        unsafe {
+            osrmc_sys::osrmc_table_params_set_annotations(
+                handle,
+                annotations.has_distance(),
```
Comment (Antti) [inline src/table.rs]: We define `Annotations` to be bitflags, but then pass them as bools. Why not just pass bitflags?

Comment (mjkillough) [reply src/table.rs]: There's no typesafe way to do that through C, so it's better to pass them as bools there.

Comment (mjkillough) [reply src/table.rs]: Although they actually are the wrong way around here. 🙈 

I'm changing the API for `libosrmc` based on some feedback and will update this.

Comment (mjkillough) [reply src/table.rs]: I've removed this from the API of `osrm-rs`, as part of reworking `libosrmc`. I think we can figure out the API for this if we ever need to _disable_ distances in the returned Table responses.

**Review by Antti — APPROVED** [review]

**Pull request merged**